frontmacs-vcs.el

```diff
@@ -1,6 +1,23 @@
 (require 'magit)
+(require 'git-gutter)
+(require 'git-gutter-fringe)
 
+;; keybindings for magit
 (global-set-key (kbd "C-x g") 'magit-status)
 (global-set-key (kbd "C-x M-g") 'magit-dispatch-popup)
 
+;; colored fringe "bars"
+(define-fringe-bitmap 'git-gutter-fr:added
+  [224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224]
+  nil nil 'center)
+(define-fringe-bitmap 'git-gutter-fr:modified
+  [224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224]
+  nil nil 'center)
+(define-fringe-bitmap 'git-gutter-fr:deleted
+  [0 0 0 0 0 0 0 0 0 0 0 0 0 128 192 224 240 248]
+  nil nil 'center)
+
+(global-git-gutter-mode +1)
+(add-hook `prog-mode-hook `git-gutter-mode)
```

Comment:
I think there might be a mis-match.

1) there are modes that aren't programming modes (markdown-mode) that should have git gutter
2) there are programming modes that exist outside a git repo that shouldn't

What if we add a hook to creating the buffer that checks if there is a git repo, and then adds it
shouldn't this be only added to buffers inside